server/image_pull.go

```diff
@@ -49,6 +83,17 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 		}
 	}
 
+	rhName, rh := s.GetRuntimeHandlerForPod(ctx, req.SandboxConfig)
+	if rh != nil {
+		if rh.RuntimePullImage {
+			// don't pull the image in crio, as it will be done by the runtime
+			log.Debugf(ctx, "Skip image pull for runtime %s - image %s", rhName, image)
+			return &types.PullImageResponse{
+				ImageRef: req.Image.Image,
```

Comment:
At least per lines just above, `req.Image` can be `nil`. And the code has already dealt with that, and the line above uses the result; i.e. the debug log and the actually-returned value are not consistent.